File: prime/first_prime.go


package main

import (
	"fmt"
	"math"
)

func is_prime (num int) bool {

	if num <= 1 {
	   return false
	}

	numF := float64 (num)
    m := math.Ceil(math.Sqrt(numF))
	l := int (m)

	for i := 2; i <= l; i++ {

		if (num % i == 0) {
			return false
		}
	}
    
	return true
}

func main () {

	fmt.Println ("First 10 prime numbers");

	numPrime := 0;
    element := 0;
    isPrime := false;
    

	for  {
           
		isPrime = is_prime (element);
		if (isPrime) {
		    fmt.Println (element)
			numPrime++;
		}	

		element++

		if (numPrime > 10) {
		    break;
		}	
            
	}


}

File: prime/first_prime.py


import math 

def is_prime (num):

    if num <= 1:
        return False

    k = math.sqrt(num)

    k = int (math.ceil(k))

    for i in range (2,k+1):
        #print (i)
        if num % i == 0:
            return False 
    return True

element = 0
numPrime = 0

while (True):
     
     isPrime = is_prime (element)
     if isPrime:
        print (element)
        numPrime = numPrime + 1  
          
     if numPrime == 10:
         break;
     element = element + 1

    





